pkg/proxy/userspace/proxier.go

```diff
@@ -450,12 +452,16 @@ func (proxier *Proxier) mergeService(service *api.Service) sets.String {
 		info.loadBalancerStatus = *helper.LoadBalancerStatusDeepCopy(&service.Status.LoadBalancer)
 		info.nodePort = int(servicePort.NodePort)
 		info.sessionAffinityType = service.Spec.SessionAffinity
+		// Use session affinity timeout value
+		if service.Spec.ClientIPAffinityConfig != nil && service.Spec.ClientIPAffinityConfig.TimeoutSeconds > 0 {
```

Comment:
API code should ensure that by the time we get here, we already can assume that if affinity == ClientIP, the config structure is non-nil